packages/core/src/shared/globalState.ts

```diff
@@ -227,6 +228,53 @@ export class GlobalState implements vscode.Memento {
         return all?.[warehouseArn]
     }
 
+    /**
+     * Get the codewhisperer customization. If legacy (map of customizations) store the
+     * customization with label of profile name
+     *
+     * @param profileName name of profile, only used in case legacy customization is found
+     * @returns codewhisperer customization, or undefined if not found.
+     * If legacy, return the codewhisperer customization for the auth profile name
+     */
+    getCodewhispererCustomization(profileName: string): Customization | undefined {
+        const result = this.tryGet('CODEWHISPERER_SELECTED_CUSTOMIZATION', Object, undefined)
+
+        // Legacy migration for old customization map of type { [label: string]: Customization[] }
+        if (typeof result === 'object' && Object.values(result).every(Array.isArray)) {
+            const selectedCustomization = result[profileName]
+            this.tryUpdate('CODEWHISPERER_SELECTED_CUSTOMIZATION', selectedCustomization)
+            return selectedCustomization
+        } else {
+            return result
+        }
+    }
+
+    /**
+     * Get the codewhisperer persisted customizations. If legacy (map of customizations) store the
+     * customizations with label of profile name
+     *
+     * @param profileName name of profile, only used in case legacy customization is found
+     * @returns array of codewhisperer persisted customizations, or empty array if not found.
+     * If legacy, return the codewhisperer persisted customizations for the auth profile name
+     */
+    getCodewhispererPersistedCCustomization(profileName: string): Customization[] {
```

Comment:
are these basically "cached" data or what is meant by persisted? if it's just a cache we should call it that.

```suggestion
    getAmazonQCachedCustomizations(profileName: string): Customization[] {
```